src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/util/RestHandlerUtils.kt

```diff
@@ -30,3 +31,26 @@ const val REFRESH = "refresh"
 
 const val WITH_TYPE = "with_type"
 val XCONTENT_WITHOUT_TYPE = ToXContent.MapParams(mapOf(WITH_TYPE to "false"))
+
+const val FAILURES = "failures"
+const val FAILED_INDICES = "failed_indices"
+const val UPDATED_INDICES = "updated_indices"
+
+fun buildInvalidIndexResponse(builder: XContentBuilder, failedIndices: MutableList<FailedIndex>) {
+    if (failedIndices.isNotEmpty()) {
+        builder.field(FAILURES, true)
+        builder.startArray(FAILED_INDICES)
+        for (failedIndex in failedIndices) {
+            builder.startObject()
+            builder.field("index_name", failedIndex.name)
```

Comment:
This could go in FailedIndex so it's reusable

Reply:
Defined ```toXContent()``` method for FailedIndex to clean this up